monitoring_service/server.py

```diff
@@ -222,23 +222,26 @@ def on_channel_close(self, event: Dict, tx: Dict):
         assert tx_balance_proof is not None
         assert is_address(closing_participant)
         assert is_channel_identifier(channel_id)
-        if channel_id not in self.state_db.monitor_requests:
-            return
-        monitor_request = self.state_db.monitor_requests[channel_id]
-        # submit monitor request
-        self.start_task(
-            OnChannelClose(self.monitor_contract, monitor_request, self.private_key),
-        )
+
+        pkey_to_mr = self.state_db.get_monitor_requests(channel_id)
+        for (_, non_closing_signer), monitor_request in pkey_to_mr.items():
+            if non_closing_signer == closing_participant:
+                # we don't have to act on behalf of the closing participant
+                continue
+            # submit monitor request
+            self.start_task(
+                OnChannelClose(self.monitor_contract, monitor_request, self.private_key),
+            )
         self.open_channels.discard(channel_id)
 
     def on_channel_settled(self, event: Dict, tx: Dict):
         channel_id = event['args']['channel_identifier']
-        monitor_request = self.state_db.monitor_requests.get(channel_id, None)
-        if monitor_request is None:
-            return
-        self.start_task(
-            OnChannelSettle(monitor_request, self.monitor_contract, self.private_key),
-        )
+        # TODO: only claim rewards if MS has submitted a BP.
+        # See https://github.com/raiden-network/raiden-monitoring-service/issues/43
+        for monitor_request in self.state_db.get_monitor_requests(channel_id).values():
```

Comment:
Can we claim every monitor request or just the latest one?

Reply:
Only the latest one. I've clarified that in the linked ticket.